src/pages/Game.tsx

```diff
@@ -1,12 +1,230 @@
+  const opponents = [];
+  const directions = [RenderDirection.LEFT, RenderDirection.TOP, RenderDirection.RIGHT];
+  let currentIndex = indexOfCurrentUser + 1;
+  for (let i = 0; i < users.length - 1; i += 1) {
+    if (currentIndex >= users.length) {
+      currentIndex = 0;
+    }
+    const opponent = new MahjongOpponent(users[currentIndex].username, directions[i]);
+    opponents.push(opponent);
+    const opponentContainer = opponent.getContainer();
+    pixiStage.addChild(opponentContainer);
+    currentIndex += 1;
+  }
+  [opponentOne, opponentTwo, opponentThree] = [opponents[0], opponents[1], opponents[2]];
+  player = new MahjongPlayer(currentUser.username);
+};
+
+/**
+ * Game Page for main game
+ */
+const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
+  const canvasRef = useRef<HTMLDivElement>(null);
+
+  const [redrawPending, setRedrawPending] = useState(false);
+
+  /**
+   * Allows animate to redraw the canvas if there is a state change.
+   */
+  const requestRedraw = () => {
+    setRedrawPending(false);
+  };
+
+  /**
+   * Game reference is passed from GameLobby through history.push
+   */
+  const history = useHistory();
+  const { state } = history.location;
+  const { game } = state as { game: Game };
+
+  /**
+   * Main Animation loop
+   */
+  function animate() {
+    if (!redrawPending) {
+      setRedrawPending(true);
+      stage.removeChildren(0, stage.children.length);
+
+      const mahjongPlayer = player as MahjongPlayer;
+      mahjongPlayer.removeAllAssets();
+      mahjongPlayer.render(spriteFactory, stage, requestRedraw);
+      mahjongPlayer.reposition(pixiApplication.view);
+
+      const mahjongOpponentOne = opponentOne as MahjongOpponent;
+      mahjongOpponentOne.removeAllAssets();
+      mahjongOpponentOne.render(spriteFactory, stage);
+      mahjongOpponentOne.reposition(pixiApplication.view);
+
+      const mahjongOpponentTwo = opponentTwo as MahjongOpponent;
+      mahjongOpponentTwo.removeAllAssets();
+      mahjongOpponentTwo.render(spriteFactory, stage);
+      mahjongOpponentTwo.reposition(pixiApplication.view);
+
+      const mahjongOpponentThree = opponentThree as MahjongOpponent;
+      mahjongOpponentThree.removeAllAssets();
+      mahjongOpponentThree.render(spriteFactory, stage);
+      mahjongOpponentThree.reposition(pixiApplication.view);
+
+      pixiApplication.render();
+    }
+    requestAnimationFrame(animate);
+  }
+
+  /**
+   * Listener Callbacks
+   */
+
+  /**
+   * For GAME_PAGE_LOAD
+   * @param payload GamePageLoadJSON
+   */
+  const confirmGamePageLoadReceived = (payload: unknown): void => {
+    const data = payload as GamePageLoadJSON;
+    const { success, error } = data;
+    if (success) {
+      ws?.removeListener(IncomingAction.GAME_PAGE_LOAD);
+    } else {
+      console.log(`Error processing GAME_PAGE_LOAD: ${error}`);
+      ws?.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { success: true });
```

Comment:
do you mean `success: false`?
```js
ws?.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { success: false });
```

Reply:
Does the backend care that the payload: success is true or false

Reply:
actually I need `gameId` as the payload for `GAME_PAGE_LOAD`
```
ws?.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { gameId: game.gameId });
```

Reply:
and these payloads for these two routes
```
DRAW_TILE: { gameId: 'game Id' }
PLAY_TILE: { gameId: 'game Id', tile: '1_DOT' }
```

Reply:
naming convention is same for inbound and outbound messages?
e.g. FrontEnd -> DRAW_TILE -> Backend -> DRAW_TILE -> FrontEnd

Reply:
Nvm, I see it on the updated UML diagram

Reply:
yes